README.md

```diff
@@ -1,3 +1,134 @@
+our first `fuse` system at this point. Below is the constraint graph that would be generated. The large circles
+represent state variables at a given time, while the small squares represent measurements. The graph connectivity
+indicates which variables are involved in what measurements.
+
+![fuse graph](doc/fuse_graph_1.png)
+
+However, typically the laser measurements and the wheel encoder measurements are not synchronized. The encoder
+measurements are sampled faster than the laser, and are sampled at different times using a different clock. In such a
+situation, the constraint graph becomes disconnected.
+
+![fuse graph](doc/fuse_graph_2.png)
+
+This is where motion models come into play. A motion model differs from a sensor model in that constraints can be
+generated between any two requested timestamps. Motion model constraints are generated upon request, not due to their
+own internal clock. We use the motion model to connect the states introduced by the sensor measurements. So we derive
+a class from the `fuse_core::MotionModel` base class that implements our differential drive kinematic constraints.
+
+![fuse graph](doc/fuse_graph_3.png)
+
+Note that there is nothing preventing the system from using multiple motion models at the same time, if that is of
+benefit.
+
+The two sensor models and the motion model are configured as plugins for an optimizer implementation. The optimizer
+handles all of the coordination between the sensors and motion models, as well as performs the required computation
+to generate the optimal state variable values.
+
+![fuse graph](doc/fuse_graph_4.png)
+
+While our `fuse` system is now optimizing constraints from two different sensors and a motion model, it is not yet
+publishing any data back out to ROS. We should probably do that. So we would derive a `fuse_core::Publisher` class
+that finds the most recent position and orientation and publishes that as a `geometry_msgs::PoseStamped` message to
+the topic of your choice.
+
+![fuse graph](doc/fuse_graph_5.png)
+
+We finally have something that is starting to be useful.
+
+### Adaptation #1: Full path publishing
+
+Nothing about the `fuse` framework limits you to having a single publisher. What if you want to visualize the entire
+robot trajectory, instead of just the most recent pose? Well, we can create a new derived `fuse_core::Publisher` class
+that publishes all of the robot poses using a `nav_msgs::Path` message.
+
+![fuse graph](doc/fuse_graph_adaptation1.png)
+
+### Adaptation #2: Changing kinematics
+
+In your spare time, you also build [autonomous power wheels racers](http://www.powerracingseries.org/). But race cars
+don't use differential drive; you need a different motion model. Easy enough. We simply derive a new
+`fuse_core::MotionModel` class that implements an Ackermann steering model. Everything else can be reused.
+
+![fuse graph](doc/fuse_graph_adaptation2.png)
```

Comment:
In the above diagrams, your MotionModel is "2d kinematic model" but the constraint graph labels those arcs as "motion model." Here you change the MotionModel to "Ackermann model" and the constraint graph labels are now "Ackermann". That might be an argument for changing the earlier arc labels.